apps/extension/src/Approvals/ConfirmSignature.tsx

```diff
@@ -0,0 +1,100 @@
+import { useCallback, useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+
+import { ActionButton, Alert, Input, Stack } from "@namada/components";
+import { shortenAddress } from "@namada/utils";
+import { SignatureDetails, Status } from "Approvals/Approvals";
+import { SubmitApprovedSignatureMsg } from "background/approvals";
+import { UnlockVaultMsg } from "background/vault";
+import { useRequester } from "hooks/useRequester";
+import { Ports } from "router";
+import { closeCurrentTab } from "utils";
+
+type Props = {
+  details?: SignatureDetails;
+};
+
+export const ConfirmSignature: React.FC<Props> = ({ details }) => {
+  const { msgId, signer } = details || {};
+
+  const navigate = useNavigate();
+  const requester = useRequester();
+  const [password, setPassword] = useState("");
+  const [error, setError] = useState<string>();
+  const [status, setStatus] = useState<Status>();
+  const [statusInfo, setStatusInfo] = useState("");
+
+  const handleApproveSignature = useCallback(async (): Promise<void> => {
+    if (!msgId || !signer) {
+      throw new Error("Not all required arguments were provided!");
+    }
+
+    setStatus(Status.Pending);
+    setStatusInfo(
+      `Decrypting keys and signing with ${shortenAddress(signer, 24)}`
+    );
+
+    try {
+      const isAuthenticated = await requester.sendMessage(
+        Ports.Background,
+        new UnlockVaultMsg(password)
+      );
+
+      if (!isAuthenticated) {
+        throw new Error("Invalid password!");
+      }
+
+      await requester
+        .sendMessage(
+          Ports.Background,
+          new SubmitApprovedSignatureMsg(signer, msgId)
```

Comment:
These are also the wrong way round, so they cancel each other out and the signing still works :smile:

Reply:
Thanks, I thought I had fixed that, but you're right! I'll fix this up now